cob_teleop/ros/src/cob_teleop.cpp

```diff
@@ -262,59 +282,55 @@ sensor_msgs::JoyFeedbackArray CobTeleop::switch_mode(){
   }
 
   std::string saytext;
-  cob_light::LightMode light;
-  light.timeout = 1;
-
-  if (mode_ == 1){
+  if (mode_ == 1)
+  {
     ROS_INFO("Switched to mode 1: move the base using twist controller");
-     saytext = "Base mode";
-     light.pulses = 1;
-  }if (mode_ == 2){
+    saytext = "Base mode";
+  }
+  if (mode_ == 2)
+  {
     ROS_INFO("Switched to mode 2: move the actuators to a default position (Trajectory controller)");
-     saytext = "Default position mode";
-     light.pulses = 2;
-  }if(mode_ == 3){
+    saytext = "Default position mode";
+  }
+  if(mode_ == 3)
+  {
     ROS_INFO("Switched to mode 3: move the actuators using joint group velocity controller");
-     saytext = "Velocity mode";
-     light.pulses = 3;
-  }if(mode_ == 4){
+    saytext = "Velocity mode";
+  }
+  if(mode_ == 4)
+  {
     ROS_INFO("Switched to mode 4: move the actuators in cartesian mode using twist controller");
-     saytext = "Cartesian mode";
-     light.pulses = 4;
+    saytext = "Cartesian mode";
   }
 
-  light.mode = 2;
-  light.frequency = 5;
-  std_msgs::ColorRGBA color;
-  color.r = 0;
-  color.g = 1;
-  color.b = 0;
-  color.a = 1;
-  light.pulses = mode_;
-  light.color = color;
-  lightgoal.mode = light;
-//  setlightclient_->sendGoal(lightgoal);
   say(saytext, false);
+  setLight(mode_);
```

Comment:
This is more an aesthetical issue:
Do we want to switch the light **after** the robot has said what it is about to say (which could take a few seconds), or **before**?

I'd say before, but this is just personal preference ;-)